### PR TITLE
feat: closes #13 and implements preview endpoint and logo via base64 in json body arg 'logo'

### LINKED ISSUE #13: Preview Endpoint
It would be great to have a dedicated endpoint for QR code generation, so that whenever we use the preview feature, we're not generating short links.

### PR DESCRIPTION
**Pull request by bolovsky**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -62,7 +62,7 @@ EOF
 # it will overlay the logo on qrcode center
 
 curl -XPOST https://yourdomain.something/api/v1/short \
-  -H 'token: ThisIsA5uper$ecureAPIToken' \
+  -H 'token: Whatever_Token_you_put_in_your_env' \
```
Comment (pedrodsrodrigues) [inline README.md:65]: If we change the value here, maybe we should change it in the other places also - to maintain consistency.

File: README.md [hunk 2]
```diff
@@ -73,7 +73,6 @@ This would render an answer like:
   "id":"4b677dfe-e17a-46e7-9cd2-25a45e8cb19c",
   "link":"https://www.google.pt/",
   "TTL":"2023-03-25T23:59:59Z",
-  "redirection_limit": 5,
```
[the remark this answers is not in the document]
Comment (bolovsky) [reply README.md:76]: hum, merge conflict, apparently I dropped that 1

File: README.md (hunk 1 above)
Comment (bolovsky) [reply README.md:65]: hum, this was already there, it was due to merge conflict, but yes, will fix it :)

File: api/response/response.go [hunk 3]
```diff
@@ -13,9 +13,9 @@ type ErrorResponse struct {
 }
 
 func NewErrorResponse(message string) *ErrorResponse {
```
[the remark this answers is not in the document]
Comment (bolovsky) [reply api/response/response.go:15]: It does, but if it is `NewError` it will be confusing in its usage. Can refactor to call it `Failure` maybe?

**Review by pedrodsrodrigues — APPROVED** [review]

**Review by pedrodsrodrigues — APPROVED** [review]

**Pull request merged**